packages/fungible-token/bridge-fungible-token/src/data_structures/deposit_message.sw

```diff
@@ -0,0 +1,67 @@
+library;
+
+use std::inputs::input_message_data;
+use ::data_structures::constants::{
+    CONTRACT_DEPOSIT,
+    CONTRACT_DEPOSIT_WITH_DATA,
+    DEPOSIT,
+    OFFSET_AMOUNT,
+    OFFSET_DECIMALS,
+    OFFSET_FROM,
+    OFFSET_MESSAGE_TYPE,
+    OFFSET_TO,
+    OFFSET_TOKEN_ADDRESS,
+    OFFSET_TOKEN_ID,
+};
+pub enum DepositType {
+    Addr: (),
+    Contract: (),
+    ContractWithData: (),
+}
+pub struct DepositMessage {
+    pub amount: b256,
+    pub from: b256,
+    pub to: Identity,
+    pub token_address: b256,
+    pub token_id: b256,
+    pub decimals: u8,
+    pub deposit_type: DepositType,
+}
+impl DepositMessage {
+    /// Read the bytes passed as message data into an in-memory representation using the DepositMessage type
+    pub fn parse_deposit_to_address(msg_idx: u64) -> Self {
+        Self {
+            amount: input_message_data(msg_idx, OFFSET_AMOUNT).into(),
+            from: input_message_data(msg_idx, OFFSET_FROM).into(),
+            token_address: input_message_data(msg_idx, OFFSET_TOKEN_ADDRESS).into(),
+            to: Identity::Address(Address::from(b256::from(input_message_data(msg_idx, OFFSET_TO)))),
+            token_id: input_message_data(msg_idx, OFFSET_TOKEN_ID).into(),
+            decimals: input_message_data(msg_idx, OFFSET_DECIMALS).get(31).unwrap(),
+            deposit_type: DepositType::Addr,
+        }
+    }
+    // /// Read the bytes passed as message data into an in-memory representation using the DepositMessage type
+    pub fn parse_deposit_to_contract(msg_idx: u64) -> Self {
+        Self {
+            amount: input_message_data(msg_idx, OFFSET_AMOUNT).into(),
+            from: input_message_data(msg_idx, OFFSET_FROM).into(),
+            token_address: input_message_data(msg_idx, OFFSET_TOKEN_ADDRESS).into(),
+            to: Identity::ContractId(ContractId::from(b256::from(input_message_data(msg_idx, OFFSET_TO)))),
+            token_id: input_message_data(msg_idx, OFFSET_TOKEN_ID).into(),
+            decimals: input_message_data(msg_idx, OFFSET_DECIMALS).get(31).unwrap(),
+            deposit_type: DepositType::Contract,
+        }
+    }
+    // /// Read the bytes passed as message data into an in-memory representation using the DepositMessage type
+    pub fn parse_deposit_to_contract_with_data(msg_idx: u64) -> Self {
```

Comment:
I went to implement it this way, but I remembered the reason to opt for the current implementation. Pure gas golfing: since I already do `match` in `message_data.sw` and kind of have this info, I avoid doing a comparison again while parsing to define the `to` type. It ends up with bigger bytecode, but performs one less computation.

Wdyt? Do we prioritise readability?

Reply:
No, I think gas golfing is better here